contracts/governance/Staking/Staking.sol

```diff
@@ -298,21 +298,113 @@ contract Staking is
     }
 
     /**
-     * @notice Withdraw tokens for vesting contract.
-     * @param vesting The address of Vesting contract.
+     * @notice Withdraw the given amount of tokens.
+     * @param amount The number of tokens to withdraw.
+     * @param until The date until which the tokens were staked.
      * @param receiver The receiver of the tokens. If not specified, send to the msg.sender
-     * @dev Can be invoked only by whitelisted contract passed to governanceWithdrawVesting.
+     * @param vestingConfig The vesting config.
+     * @dev Only used to governance direct withdrawal.
      * */
-    function governanceWithdrawVesting(address vesting, address receiver)
-        public
-        onlyAuthorized
-        whenNotFrozen
-    {
-        vestingWhitelist[vesting] = true;
-        ITeamVesting(vesting).governanceWithdrawTokens(receiver);
-        vestingWhitelist[vesting] = false;
+    function _governanceWithdrawDirect(
+        uint96 amount,
+        uint256 until,
+        address receiver,
+        VestingConfig memory vestingConfig
+    ) internal {
+        _notSameBlockAsStakingCheckpoint(until);
+
+        _withdrawDirect(amount, until, receiver, vestingConfig);
+        // @dev withdraws tokens for lock date 2 weeks later than given lock date if sender is a contract
+        //		we don't need to check block.timestamp here
+        _withdrawNextDirect(until, receiver, vestingConfig);
+    }
+
+    /**
+     * @notice New governance withdraw vesting directly through staking contract.
+     * This direct withdraw vesting, can solve the out of gas issue from the old withdraw vesting function.
+     * This function only allowing the call from vesting contract which type is TeamVesting.
+     *
+     * @param vesting The vesting address.
+     * @param receiver The receiving address.
+     * @param startFrom The start value for the iterations.
+     */
+    function governanceDirectWithdrawVesting(
+        address vesting,
+        address receiver,
+        uint256 startFrom
+    ) external onlyAuthorized whenNotFrozen {
+        _governanceDirectWithdrawVesting(vesting, receiver, startFrom);
 
-        emit VestingTokensWithdrawn(vesting, receiver);
+        emit VestingTokensWithdrawn(vesting, receiver, startFrom);
```

Comment:
event VestingTokensWithdrawn is superfluous and can be removed
we already have proper events in _governanceDirectWithdrawVesting: 
IncompleteGovernanceWithdrawTokens and TokensWithdrawn

Reply:
fixed in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/447/commits/892d1be815121b95033677706a1c9cb5677c31e7)